xcdat/regridder/xgcm.py

```diff
@@ -210,6 +213,26 @@ def vertical(self, data_var: str, ds: xr.Dataset) -> xr.Dataset:
 
         return output_ds
 
+    def _infer_target_data(self, ds) -> xr.DataArray | None:
```

Comment:
I just rebased this branch on the latest `main` with PR #721. 

On this specific line, I removed the use of Union with |.